client/widgets/CArtifactHolder.cpp

```diff
@@ -654,7 +654,7 @@ CArtifactsOfHero::~CArtifactsOfHero()
 		}
 		else
 		{
-			//TODO remove artifact
+			LOCPLINT->cb->eraseArtifactByClient(ArtifactLocation(curHero, ArtifactPosition::TRANSITION_POS));
```

Comment:
To clarify - this happens if we exit hero window with picked up artifact and have no place to put artifact down, e.g. full backpack. Correct?

Reply:
Right. Easy to reproduce with war machines. Pick up the war machine on the exchange screen. And do swap on another hero's war machine. Close the window without placing art in the slot.